src/nvim/ui.c

```diff
@@ -148,7 +148,11 @@ bool ui_rgb_attached(void)
     return true;
   }
   for (size_t i = 0; i < ui_count; i++) {
-    if (uis[i]->rgb) {
+    // We do not consider the TUI in this loop because we already checked for 'termguicolors' at the
+    // beginning of this function. In this loop, we are checking to see if any _other_ UIs which
+    // support RGB are attached.
+    bool tui = uis[i]->stdin_tty || uis[i]->stdout_tty;
+    if (!tui && uis[i]->rgb) {
       return true;
     }
```

Comment:
Since `p_tgc` is not checked above in headless mode, I think extra changes will be needed for headless mode. Something like this:
```suggestion
    }
    if (tui && p_tgc) {
      return true;
    }
```

Reply:
or maybe just change the check on the top to just check for `p_tgc` regardless of `headless_mode`. in headless mode 'termguicolors' can only be true if explicitly set, in case it conveys the user's explicit intent.

Reply:
Oh, right.

Reply:
a55dbfe0125fb7c7a049130ef42a375a66199061